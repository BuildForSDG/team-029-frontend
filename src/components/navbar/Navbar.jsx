import React from 'react';
import { bool, string } from 'prop-types';
import { Link } from 'react-router-dom';
import logo from '../../assets/logo.png';

import './navbar.styles.scss';

const Navbar = ({ hasLinks, activeLink }) => {
	return (
		<div className="nav-container">
			<div className="logo-container">
				<Link to="/"><img src={logo} alt="logo" /></Link>
			</div>
			{hasLinks && (
				<ul>
					<li className={activeLink === 'Home' ? 'active' : ''}>
						<Link to="/">Home</Link>
					</li>
					<li className={activeLink === 'Road Info' ? 'active' : ''}>
						<Link to="/">Road Info</Link>
					</li>
					<li className={activeLink === 'Login Admin' ? 'active' : ''}>
						<Link to="/login">Login</Link>
					</li>
				</ul>
			)}
		</div>
	);
};

Navbar.defaultProps = {
	hasLinks: true,
	activeLink: 'Home'
};

Navbar.propTypes = {
	hasLinks: bool,
	activeLink: string
};

export default Navbar;
