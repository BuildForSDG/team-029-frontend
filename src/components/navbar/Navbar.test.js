import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Navbar from './Navbar.jsx';

afterEach(cleanup);

it('Should render <Navbar />', () => {
	const { asFragment } = render(<Navbar />);
	expect(asFragment()).toMatchSnapshot();
});
