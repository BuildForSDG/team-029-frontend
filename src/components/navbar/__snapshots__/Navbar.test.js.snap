// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render <Navbar /> 1`] = `
<DocumentFragment>
  <div
    class="nav-container"
  >
    <div
      class="logo-container"
    >
      <img
        alt="logo"
        src="logo.png"
      />
    </div>
    <ul>
      <li
        class="active"
      >
        <a>
          Home
        </a>
      </li>
      <li
        class=""
      >
        <a>
          Road Info
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
